### ISSUE TITLE
Make it work for Macbook pro M1 with GPU / metal support

### ISSUE THREAD

**Issue by ejmiddle** [marella/ctransformers#97]
Hi, I am trying to make work a metal supported inference on my macos M1 2021 with the  "TheBloke/Llama-2-7B-Chat-GGML" model. 

I installed from source as described and this is essentially the code I run:

from ctransformers import AutoModelForCausalLM
model = AutoModelForCausalLM.from_pretrained("TheBloke/Llama-2-7B-Chat-GGML")
print(model('AI is going to'))

It works with CPU support, however, it is as slow as it was before activating metal support. Approx 10 - 20 seconds (16GB RAM).
 
Adding gpu_layers as below seems to be a command for cuda only, hence I get an error to install ctransformers with CT_CUBLAS=1, which is sth that does not make sense on MacOS!?
model = AutoModelForCausalLM.from_pretrained("TheBloke/Llama-2-7B-Chat-GGML" , gpu_layers=1)

pytorch is properly installed and test to see mps is properly activated worked.

What did I miss? How can I check if the installation with CT_METAL = 1 actually worked (environment variable is there)? How can I force mps support? Any ideas? 

Would love to stick with this approach to make it run.

Thanks
Andi

**Comment by marella** [comment]
Hi, setting `gpu_layers=1` is required to run on Metal. If you are seeing CUBLAS error then it was not installed correctly with Metal enabled. I will change the error message for MacOS.

Try uninstalling and installing again and post the output:

```sh
pip uninstall ctransformers --yes
CT_METAL=1 pip install ctransformers --no-binary ctransformers
```

**Comment by ejmiddle** [comment]
Hi Marella, thx for pointing out what the problem was. I made it work now. 

I did not manage to install it properly with pipenv (tried configuring the Pipfile accordingly and it installed successfully, but apparently doesn't use the env variable or no binary instruction). With another virtual environment manager (venv) it worked immediately.

One more question: Does the number of layers have an effect also for metal? Or is it either on or off?
